File: src/main/java/com/flaghacker/uttt/bots/RandomBot.java
package com.flaghacker.uttt.bots;

import com.flaghacker.uttt.common.AbstractBot;
import com.flaghacker.uttt.common.Board;
import com.flaghacker.uttt.common.Coord;
import com.flaghacker.uttt.common.Util;

import java.util.List;
import java.util.Random;

public class RandomBot extends AbstractBot
{
	private Random random = Util.loggedRandom();

	@Override
	public Coord move(Board board)
	{
		List<Coord> moves = board.availableMoves();
		return moves.get(random.nextInt(moves.size()));
	}

	@Override
	public String toString()
	{
		return "RandomBot";
	}
}

File: src/main/kotlin/bots/RandomBot.kt
package bots

import common.Board
import common.Bot
import common.Coord
import java.util.*

class RandomBot(private val random: Random = Random()) : Bot {
	override fun move(board: Board, percentDone: (Int) -> Unit): Coord {
		percentDone(100)
		val moves = board.availableMoves
		if (moves.isEmpty()) throw Exception("No moves remain")
		return moves[random.nextInt(moves.size)]
	}

	override fun reset() {}
	override fun cancel() {}
	override fun toString() = "RandomBot"
}
